src/components/themed/WalletListCurrencyRow.js

```diff
@@ -1,228 +1,203 @@
+
+  const [{ fiatText: fiatBalanceText }] = useFiatText({
+    nativeCryptoAmount: balance,
+    cryptoCurrencyCode: currencyCode,
+    isoFiatCurrencyCode,
+    cryptoExchangeMultiplier
+  })
 
-    const exchangeData = {
-      exchangeRateString: '',
-      exchangeRateType: 'Neutral'
-    }
-    if (showRate) {
-      const { differencePercentageStyle, differencePercentageString, exchangeRateFiatSymbol, exchangeRateString } = this.getRate()
-      exchangeData.exchangeRateString = `${exchangeRateFiatSymbol}${exchangeRateString} ${differencePercentageString}`
-      exchangeData.exchangeRateType = differencePercentageStyle
-    }
+  let [{ fiatText: exchangeRateText }] = useFiatText({
+    nativeCryptoAmount: cryptoExchangeMultiplier,
+    cryptoCurrencyCode: currencyCode,
+    isoFiatCurrencyCode,
+    cryptoExchangeMultiplier
+  })
+  let exchangeRateType = 'neutral'
 
-    return (
-      <CardContent
-        image={this.renderIcon()}
-        title={
-          <>
-            <EdgeText style={styles.detailsCurrency}>{currencyCode}</EdgeText>
-            <EdgeText style={[styles.detailsExchange, styles[`percentage${exchangeData.exchangeRateType}`]]}>{exchangeData.exchangeRateString}</EdgeText>
-          </>
-        }
-        value={<EdgeText style={styles.detailsValue}>{cryptoAmount}</EdgeText>}
-        subTitle={walletNameString}
-        subValue={fiatBalanceSymbol + fiatBalanceString}
-      />
-    )
+  if (showRate) {
+    const { differencePercentageString, differencePercentageStyle } = getDifference({
+      exchangeRate,
+      currencyCode,
+      exchangeRates,
+      fiatExchangeRate: isoFiatCurrencyCode !== 'iso:USD' ? exchangeRates[`iso:USD_${isoFiatCurrencyCode}`] : '1'
+    })
+    exchangeRateText += ` ${differencePercentageString}`
+    exchangeRateType = differencePercentageStyle
   }
 
-  render() {
-    const { onPress, onLongPress, paddingRem, gradient } = this.props
-    return (
-      <ClickableRow onPress={onPress || this.noOnPress} onLongPress={onLongPress} gradient={gradient} paddingRem={paddingRem} highlight>
-        {this.renderChildren()}
-      </ClickableRow>
-    )
-  }
+  const icon = useMemo(() => <WalletProgressIcon currencyCode={currencyCode} walletId={walletId} />, [currencyCode, walletId])
+  const children = useMemo(
+    () => (
+      <View style={styles.balance}>
+        <EdgeText>{nativeCryptoAmount}</EdgeText>
+        <EdgeText style={styles.fiatBalance}>{fiatBalanceText}</EdgeText>
+      </View>
+    ),
+    [fiatBalanceText, nativeCryptoAmount, styles.balance, styles.fiatBalance]
+  )
+
+  return (
+    <WalletListRow
+      currencyCode={currencyCode}
+      exchangeRateText={showRate ? exchangeRateText : null}
+      exchangeRateType={showRate ? exchangeRateType : null}
+      icon={icon}
+      onPress={onPress}
+      onLongPress={onLongPress}
+      walletName={walletName ?? name ?? `My ${getCurrencyInfo(allCurrencyInfos, currencyCode)?.displayName ?? ''}`}
```

Comment:
We can ditch `getCurrencyInfo` and `allCurrencyInfos`, since it's just `edgeWallet.currencyInfo`.